components/event-sources/adapter/http/adapter.go

```diff
@@ -0,0 +1,196 @@
+package http
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	cloudevents "github.com/cloudevents/sdk-go"
+	"github.com/kyma-project/kyma/components/event-sources/apis/sources"
+	"go.uber.org/zap"
+	"knative.dev/eventing/pkg/adapter"
+	"knative.dev/pkg/logging"
+	"knative.dev/pkg/source"
+	"log"
+	"net/http"
+)
+
+type envConfig struct {
+	adapter.EnvConfig
+	EventSource string `envconfig:"EVENT_SOURCE" required:"true"`
+
+	// PORT as required by knative serving runtime contract
+	Port int `envconfig:"PORT" required:"true" default:"8080"`
+}
+
+func (e *envConfig) GetSource() string {
+	return e.EventSource
+}
+
+func (e *envConfig) GetPort() int {
+	return e.Port
+}
+
+type httpAdapter struct {
+	ceClient       cloudevents.Client
+	statsReporter  source.StatsReporter
+	accessor       AdapterEnvConfigAccessor
+	adapterContext context.Context
+	logger         *zap.Logger
+}
+
+type AdapterEnvConfigAccessor interface {
+	adapter.EnvConfigAccessor
+	GetSource() string
+	GetPort() int
+}
+
+const resourceGroup = "http." + sources.GroupName
+
+const (
+	// endpoint for cloudevents
+	endpointCE = "/"
+	// endpoint for readiness check
+	readinessReadiness = "/healthz"
+)
+
+const (
+	ErrorResponseCEVersionUnsupported = "unsupported cloudevents version"
+	ErrorResponseSendToSinkFailed     = "unable to forward event to sink"
+)
+
+func NewEnvConfig() adapter.EnvConfigAccessor {
+	return &envConfig{}
+}
+
+func NewAdapter(ctx context.Context, processed adapter.EnvConfigAccessor, ceClient cloudevents.Client, reporter source.StatsReporter) adapter.Adapter {
+
+	accessor, ok := processed.(AdapterEnvConfigAccessor)
+
+	if !ok {
+		panic(fmt.Sprintf("cannot create adapter, expecting a *envconfig, but got a %T", processed))
+	}
+
+	return &httpAdapter{
+		adapterContext: ctx,
+		ceClient:       ceClient,
+		statsReporter:  reporter,
+		accessor:       accessor,
+		logger:         logging.FromContext(ctx).Desugar(),
```

Comment:
Any particular reason for de-sugarring the logger? The sugarred logger provides an API that's slightly more use-friendly.

Reply:
We can also use the sugared logger if you prefer. I know the desugared logger's performance is a bit better, but that is not the reason why I choose it. I am used to the desugared one from knative code.

Reply:
I'm fine with both 👍